tools/LogCollector/source/update_mgmt_health_check.py

```diff
@@ -1,268 +1,392 @@
+import subprocess
 
 from os import walk
 
-def main(output_path=None):
-    output = []
-
+rule_info_list = []
+output = []
+
+oms_admin_conf_path = "/etc/opt/microsoft/omsagent/conf/omsadmin.conf"
+oms_agent_dir = "/var/opt/microsoft/omsagent"
+oms_agent_log = "/var/opt/microsoft/omsagent/log/omsagent.log"
+current_mof = "/etc/opt/omi/conf/omsconfig/configuration/Current.mof"
+status_passed = "Passed"
+status_failed = "Failed"
+status_debug = "Debug"
+empty_failure_reason = ""
+workspace = ""
+
+class RuleInfo:
+    def __init__(self, rule_id, rule_group_id, status, result_msg_id):
+        self.RuleId = rule_id
+        self.RuleGroupId = rule_group_id
+        self.CheckResult = status
+        self.CheckResultMessageId = result_msg_id
+        self.CheckResultMessageArguments = list()
+
+def main(output_path=None, return_json_output="False"):
     if os.geteuid() != 0:
-        print "Please run this script in sudo"
+        print "Please run this script as root"
         exit()
 
-    output.append(get_machine_info() + "\n")
-    output.append(check_oms_admin() + "\n")
-    output.append(check_oms_agent_installed() + "\n")
-    output.append(check_oms_agent_running() + "\n")
-    output.append(check_hybrid_worker_installed() + "\n")
-    output.append(check_hybrid_worker_running() + "\n")
-    output.append(check_network() + "\n")
-
-    output.append("\n")
-
-    if output_path is not None:
-        try: 
-            os.makedirs(output_path)
-        except OSError:
-            if not os.path.isdir(output_path):
-                raise
-        f = open(output_path + "/healthcheck-" + str(datetime.datetime.utcnow().isoformat()) + ".log", "w")
-        f.write("".join(output))
-
-    print "".join(output)
-
-def check_endpoints(workspace, endpoints, success_message, failure_message):
-    output = []
-
-    for endpoint in endpoints:
-        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-
-        new_endpoint = None
-        
-        if "*" in endpoint and workspace is not None:
-            new_endpoint = endpoint.replace("*", workspace)
-        elif "*" not in endpoint:
-            new_endpoint = endpoint
-
-        if new_endpoint is not None:
-            try:
-                response = sock.connect_ex((new_endpoint, 443))
-
-                if response == 0:
-                    output.append(new_endpoint + ": " + success_message + "\n")
-                else:
-                    output.append(new_endpoint + ": " + failure_message + "\n")
-
-            except Exception as ex:
-                output.append(new_endpoint + ": " + failure_message + "\n")
-                output.append(str(ex) + "\n")
-
-    return output
-
-def check_network():
-    output = []
-    output.append("Network check: \n\n")
+    # supported python version 2.4.x to 2.7.x
+    if not ((sys.version_info[0] == 2) and ((sys.version_info[1]>=4) and (sys.version_info[1] < 8))):
+        print("Unsupport python version:" + str(sys.version_info))
```

Comment:
adding this to print statement will help customer -"supported python version 2.4.x to 2.7.x"

Reply:
fixed